### PR TITLE
feat: use listr and ensure changeset creation before migration export [PHX-2614]

### PR DESCRIPTION
**Pull request by marcolink**
This PR introduces 2 significant improvements. 

**Wait for changeset creation before generating migrations.** 
This is currently based on an ordinary timeout. As soon as [this](https://github.com/contentful/contentful-management.js/pull/1777) works as expected, we can switch to the newly introduced function to create and await the changeset result.   

**Use listr to show progress**
We use listr to indicate progress to the user. Depending on the command, this can show different granular information.

Screenshot 2023-05-04 at 08 15 19

### CODE REVIEW

**Review by andreascful — APPROVED** [review]
Awesome PR Marco!!

File: lib/utils/app-actions.ts [hunk 1]
```diff
@@ -53,6 +53,24 @@ export const callCreateChangeset = async ({
   return createResponse.sys.id
 }
 
+export const callCreateChangesetWithResponse = async ({
```
[the remark this answers is not in the document]
Comment (marcolink) [reply lib/utils/app-actions.ts:56]: it's a placeholder. I actually think we should wait with merge until this works as expected.

File: lib/cmds/merge_cmds/export.ts [hunk 2]
```diff
@@ -151,9 +204,12 @@ const exportEnvironmentMigration = async ({
...
   })
 
-  await writeFileP(outputTarget, migration)
-
-  success(`✅ Migration exported to ${outputTarget}.`)
+  if (result.migration) {
+    await writeFileP(outputTarget, result.migration)
+    success(`✅ Migration exported to ${outputTarget}.`)
+  } else {
+    error(`failed to save migration file!`)
```
Comment (damienxy) [inline lib/cmds/merge_cmds/export.ts:211]: Should we move this error to the `mergeErrors` file?

**Review by damienxy — APPROVED** [review]
Just tried it out and looks awesome!

**Pull request merged**